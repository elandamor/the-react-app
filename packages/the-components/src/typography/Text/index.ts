import {
  color,
  layout,
  shouldForwardProp,
  space,
  styled,
  StyledSystemProps,
  typography,
} from '@app/theme';

/**
 * @render react
 * @name Text component
 * @description Span (default)
 * @example
 *  <Text>Lorem ipsum dolor sit amet consectetur adipisicing elit.</Text>
 */

const Text = styled('span', { shouldForwardProp })<StyledSystemProps>(
  {
    color: 'inherit',
    lineHeight: 1.5,
  },
  color,
  layout,
  space,
  typography,
);

export default Text;
