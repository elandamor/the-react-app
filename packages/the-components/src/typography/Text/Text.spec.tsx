import { render } from '@testing-library/react';
import React from 'react';
import Text from './index';

describe('Text', () => {
  it('renders without crashing', () => {
    const { container } = render(<Text>Heading</Text>);
    expect(container.firstChild?.nodeName.toLowerCase()).toBe('span');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correct text', () => {
    const { getByText } = render(<Text>Heading</Text>);
    expect(getByText('Heading')).toBeTruthy();
  });
});
