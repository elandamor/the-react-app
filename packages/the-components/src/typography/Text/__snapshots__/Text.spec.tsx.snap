// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders without crashing 1`] = `
.emotion-0 {
  color: inherit;
  line-height: 1.5;
}

<span
  class="emotion-0"
>
  Heading
</span>
`;
